facade/hyper_block_facade.go

```diff
@@ -48,24 +48,44 @@ func NewHyperBlockFacade(
 }
 
 // GetHyperBlockByNonce will fetch the hyper block from Elrond proxy with provided nonce and options in covalent format
-func (hpf *hyperBlockFacade) GetHyperBlockByNonce(nonce uint64, options config.HyperBlockQueryOptions) (*api.CovalentHyperBlockApiResponse, error) {
-	blockByNoncePath := fmt.Sprintf("%s/%d", hyperBlockPathByNonce, nonce)
-	fullPath := hpf.getFullPathWithOptions(blockByNoncePath, options)
-
-	return hpf.getHyperBlock(fullPath)
+func (hbf *hyperBlockFacade) GetHyperBlockByNonce(nonce uint64, options config.HyperBlockQueryOptions) (*api.CovalentHyperBlockApiResponse, error) {
+	fullPath := hbf.getHyperBlockByNonceFullPath(nonce, options)
+	return hbf.getHyperBlock(fullPath)
 }
 
-// GetHyperBlockByHash will fetch the hyper block from Elrond proxy with provided hash and options in covalent format
-func (hpf *hyperBlockFacade) GetHyperBlockByHash(hash string, options config.HyperBlockQueryOptions) (*api.CovalentHyperBlockApiResponse, error) {
-	blockByHashPath := fmt.Sprintf("%s/%s", hyperBlockPathByHash, hash)
-	fullPath := hpf.getFullPathWithOptions(blockByHashPath, options)
+// GetHyperBlocksByInterval will fetch the hyper blocks from Elrond proxy with provided nonces interval and options in covalent format
+func (hbf *hyperBlockFacade) GetHyperBlocksByInterval(noncesInterval *api.Interval, options config.HyperBlockQueryOptions) (*api.CovalentHyperBlocksApiResponse, error) {
+	if noncesInterval.Start > noncesInterval.End {
+		return nil, errInvalidNoncesInterval
+	}
+
+	// Dummy implementation with no parallel bulk requests. This implementation will follow in next PR
+	encodedHyperBlocks := make([][]byte, 0, noncesInterval.End-noncesInterval.Start)
```

Comment:
this won't work if you want to get a single hyperblock in a bulk, for example the interval [37,37]

Reply:
update: won't work in any case. the size should be `end - start +1`. for example, [37,39] would result in a size of 2, whilst a size of 3 is needed

Reply:
It works for any case, I've also tested them before.
That's just memory allocation, not size allocation.
But you are right indeed, memory allocation is incorrect, should've used `end - start + 1`.
Refactored as suggested